dnf5/commands/repoquery/repoquery.cpp

```diff
@@ -714,6 +773,42 @@ void RepoqueryCommand::run() {
     // APPLY TRANSFORMS - these are not order independent and have to be applied last
     // They take a set of packages and turn it into a different set of packages
 
+    if (recursive->get_value() && !whatrequires->get_value().empty()) {
+        auto filter_requirers = [&](libdnf5::rpm::PackageQuery & q_in, const libdnf5::rpm::PackageSet & candidates) {
+            q_in.filter_requires(candidates);
+        };
+
+        result_query = repeat_filter(filter_requirers, result_query, arch->get_value());
```

Comment:
Minor: if we turn the `repeat_filter` into private member, you don't need to pass `arch->get_value()`.

Reply:
I actually prefer it being separate, for me it is easier to think about when I know it cannot touch anything outside the parameters.
However if you still think it would be better as a member I can do that.

Reply:
Sure, np.